### ISSUE TITLE
I'd like there to be an EFCore.Design.Specification.Tests 

### ISSUE THREAD

**Issue by NinoFloris** [dotnet/efcore#12326]
I'm currently working on https://github.com/bricelam/EFCore.FSharp/ and have this weekend started porting infra for the design time tests around the code generators. 

It would help a lot if this infra was actually readily available as a nuget package like the other `Specification.Tests` projects so tooling projects like EFCore.FSharp can be brought up to spec easier and faster.

**Comment by bricelam** [comment]
When I originally thought about this (when creating that repo), I decided it would be more trouble than it's worth. I think the best approach is to create an `IModel` that exercised all the parts of the code generator and create tests compare against a baseline output.

**Comment by bricelam** [comment]
Something like [test/EFCore.Design.Tests/Scaffolding/Internal/CSharpEntityTypeGeneratorTest.cs](https://github.com/aspnet/EntityFrameworkCore/blob/dev/test/EFCore.Design.Tests/Scaffolding/Internal/CSharpEntityTypeGeneratorTest.cs)

**Comment by ajcvickers** [comment]
Discussed this in triage and agreed with @bricelam that there doesn't seem enough value in doing this. @NinoFloris we could still be convinced otherwise by an ... er ,,, convincing argument.

**Comment by NinoFloris** [comment]
Yes what I think would help here is once we get the implementation up on EFCore.FSharp is to check what can and should ideally be shared.

**Comment by bricelam** [comment]
Yes, I imagine we can create a model with really good coverage of the code generation logic. We can share that between F#, VB, C#, etc.
